test/test-functions

```diff
@@ -2065,24 +2070,43 @@ test_setup() {
         mount_initdir
     else
         if [ ! -e "$IMAGE_PUBLIC" ]; then
-            # Create the backing public image, but then completely unmount
-            # it and drop the loopback device responsible for it, since we're
-            # going to symlink/copy the image and mount it again from
-            # elsewhere.
-            test_create_image
-            test_setup_cleanup
-            umount_loopback
-            cleanup_loopdev
+            # default.img is the base that every test uses and optionally appends to
+            if [ ! -e "${IMAGESTATEDIR}/default.img" ] || [ -n "${TEST_FORCE_NEWIMAGE}" ]; then
+                # Create the backing public image, but then completely unmount
```

Comment:
I'm confused how this works with parallelization: is there something preventing multiple jobs to create  $IMAGE_PUBLIC? Your patch doesn't seem to change anything wrt that, so we can merge it regardless.